src/consensus/sub_block_record.py

```diff
@@ -52,6 +55,13 @@ class SubBlockRecord(Streamable):
     # Sub-epoch (present iff this is the first SB after sub-epoch)
     sub_epoch_summary_included: Optional[SubEpochSummary]
 
+    # todo remove the hash lists (redundant)
+    finished_sub_slots: List[EndOfSubSlotBundle]  # If first sb
+    reward_chain_sub_block: RewardChainSubBlock  # Reward chain trunk data
```

Comment:
There is a lot of redundant data and also this drastically increases the size of the sub block records. Can you explain the rationale for adding these things to sub block record?

Reply:
If we have 32k sub-blocks in the cache, and each sub block now has 2 VDF proofs, this might be way too big to keep in memory

Reply:
same rational as with the proofs, cleaner code and no need to work with 2 data structures that represent the same object (sub_block_rec / block_header), but like we talked about we can optimize this